app/controllers/pieces_controller.rb

```diff
@@ -24,4 +33,10 @@ def current_piece
   def piece_params
     params.require(:piece).permit(:x, :y)
   end
+
+  def require_authorized_for_updating_piece
+    unless current_piece.color = color
```

Comment:
you'll want `==` here.

`=` always changes the value (and if successful returns true)

`==` compares the two values.

And I'm thinking you'll want `color` to be `current_user.color`